src/Workspaces/CSharp/Portable/Simplification/CSharpSimplificationService.Expander.cs

```diff
@@ -254,9 +254,32 @@ public override SyntaxNode VisitSimpleLambdaExpression(SimpleLambdaExpressionSyn
             public override SyntaxNode VisitArgument(ArgumentSyntax node)
             {
                 _cancellationToken.ThrowIfCancellationRequested();
+                string inferredName = null;
+                if (node.Parent.IsKind(SyntaxKind.TupleExpression))
+                {
+                    var tuple = (TupleExpressionSyntax)node.Parent;
+                    if (!IsTupleInDeconstruction(tuple))
+                    {
+                        inferredName = node.Expression.TryGetInferredMemberName().ValueText;
+                        if (!CanMakeNameExplicitInTuple(tuple, inferredName))
+                        {
+                            inferredName = null;
```

Comment:
i don't lke the "is null, then assign not null, then assign null" pattern.  Just get teh value, test it, and if it's good, assign to the outer local.